src/main/kotlin/io/sip3/salto/ce/sip/SipCallHandler.kt

```diff
@@ -93,71 +84,41 @@ open class SipCallHandler : AbstractVerticle() {
         }
 
         vertx.setPeriodic(expirationDelay) {
-            terminateExpiredTransactions()
             terminateExpiredCallSessions()
         }
 
         val index = config().getInteger("index")
-        vertx.eventBus().localConsumer<Pair<Packet, SIPMessage>>(PREFIX + "_$index") { event ->
+        vertx.eventBus().localConsumer<SipTransaction>(PREFIX + "_$index") { event ->
             try {
-                val (packet, message) = event.body()
-                handle(packet, message)
+                val transaction = event.body()
+                handle(transaction)
             } catch (e: Exception) {
                 logger.error("SipCallHandler 'handle()' failed.", e)
             }
         }
     }
 
-    open fun handle(packet: Packet, message: SIPMessage) {
-        val tid = message.transactionId()
-        when (message.cseqMethod()) {
+    open fun handle(transaction: SipTransaction) {
+        when (transaction.cseqMethod) {
             "INVITE" -> {
-                inviteTransactions.getOrPut(tid) { SipTransaction() }.apply {
-                    addMessage(packet, message)
-
-                    if (message.hasSdp()) {
-                        vertx.eventBus().send(RoutesCE.sdp_session, Pair(SdpHandler.CMD_HANDLE, message), USE_LOCAL_CODEC)
-                    }
+                if (transaction.response?.hasSdp() ?: false) {
```

Comment:
If we leave it here, how are we going to send information about early media? I guess that this code has to be in `SipTransactionHandler`